packages/website/docs/using-algolia-insights-plugin.md

```diff
@@ -3,6 +3,235 @@ id: using-algolia-insights-plugin
+        }
+      }
+    ];
+  }
+});
+
+function ProductItem({ hit }) {
+  return (
+    <Fragment>
+      <div className="aa-ItemIcon">
+        <img src={hit.image} alt={hit.name} width="40" height="40" />
+      </div>
+      <div className="aa-ItemContent">
+        <div className="aa-ItemContentTitle">
+          {snippetHit({ hit, attribute: "name" })}
+        </div>
+        <div className="aa-ItemContentDescription">
+          {snippetHit({ hit, attribute: "description" })}
+        </div>
+      </div>
+      <button
+        className="aa-ItemActionButton aa-TouchOnly aa-ActiveOnly"
+        type="button"
+        title="Select"
+      >
+        <svg fill="currentColor" viewBox="0 0 24 24" width="20" height="20">
+          <path d="M18.984 6.984h2.016v6h-15.188l3.609 3.609-1.406 1.406-6-6 6-6 1.406 1.406-3.609 3.609h13.172v-4.031z"></path>
+        </svg>
+      </button>
+    </Fragment>
+  );
+}
+```
+
+Now, whenever the autocomplete shows products in the dropdown, the plugin sends [view events](https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-and-conversion-analytics/in-depth/capturing-user-behavior-as-events/#view) with the [`eventName`](https://www.algolia.com/doc/api-reference/api-methods/viewed-object-ids/#method-param-eventname) "Items Viewed" for these products. If a user clicks a product, the plugin sends a [click event](https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-and-conversion-analytics/in-depth/capturing-user-behavior-as-events/#click), with the [`eventName`](https://www.algolia.com/doc/api-reference/api-methods/clicked-object-ids-after-search/#method-param-eventname) "Item Selected" for that product. By default, the plugin doesn't send any [conversion events](https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-and-conversion-analytics/in-depth/capturing-user-behavior-as-events/#conversion).
+
+You can change any of the plugins default behavior by using the [`onItemsChange`](createAlgoliaInsightsPlugin#onitemschange), [`onSelect`](createAlgoliaInsightsPlugin#onselect), or [`onActive`](createAlgoliaInsightsPlugin#onactive) hooks. For example, you may want to customize the `eventName` to match your [naming convetions](https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-and-conversion-analytics/in-depth/best-practices-for-sending-events/#naming-events-consistently) or send a different events on these actions.
+
+This snippet shows how to instantiate a plugin that sends a click event with "Product Selected from Autocomplete" as the `eventName` whenever a user selects an item.
+
+```js
+const appId = "latency";
+const apiKey = "6be0576ff61c053d5f9a3225e2a90f76";
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient("init", { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
+  insightsClient,
+  onSelect: {/***/}
```

Comment:
First, wanted to check if you thought this was a good example. Second, need some help with it. 😅

Reply:
Here's how you would do it, and yes, that's a good idea:

```js
const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
  insightsClient,
  onSelect({ insights, insightsEvents }) {
    const events = insightsEvents.map((insightsEvent) => ({
      ...insightsEvent,
      eventName: 'Product Selected from Autocomplete',
    }));
    insights.clickedObjectIDsAfterSearch(...events);
  },
});
```

Reply:
More generally, you can check the default implementation of the params here: https://github.com/algolia/autocomplete.js/blob/8a4da0211d9f671efe1c26f1ec6179fb4a77ff1c/packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts#L65-L73

Reply:
Thanks for this @francoischalifour ! Is there a way to send different events for different sections of the autocomplete? Say products vs. suggestions? Or will the insights plugin send the same event type /name etc. for each action (onItemsChange, OnSelect, OnActive)?

Reply:
You could do it conditionally on the index name:

```ts
const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
  insightsClient,
  onSelect({ insights, insightsEvents }) {
    const events = insightsEvents.map((insightsEvent) => {
      switch (insightsEvent.index) {
        case 'instant_search_demo_query_suggestions': {
          return {
            ...insightsEvent,
            eventName: 'Item Selected from Query Suggestions',
          };
        }
        case 'instant_search': {
          return {
            ...insightsEvent,
            eventName: 'Item Selected from Products',
          };
        }
        default: {
          return {
            ...insightsEvent,
            eventName: 'Item Selected from Autocomplete',
          };
        }
      }
    });
    insights.clickedObjectIDsAfterSearch(...events);
  },
});
```